server/app/analytics/job.py

```diff
@@ -0,0 +1,93 @@
+from google.appengine.ext import ndb, deferred
+from app.analytics.mapper import Mapper
+from app.models import Base
+
+
+class JobMapper(Mapper):
+    """
+    A wrapper class for the NDB Mapper that stores
+    map results in an NDB AnalyticsDump in addition to just mapping.
+    """
+    def __init__(self, kind, job_dump, job_mapper, filters):
+        super(JobMapper, self).__init__()
+        self.kind = kind
+        self.job_dump = job_dump
+        self.job_mapper = job_mapper
+        self.set_filters(filters)
+
+    def map(self, entity):
+        try:
+            map_result = self.job_mapper(entity)
+        except:
+            raise deferred.PermanentTaskFailure()
+        self.job_dump.map_result.append(map_result)
```

Comment:
Yea, so I'm saving the results right now for debugging purposes and I'll add the option to make it optional to store those results.